### DISCUSSION TITLE
Mesh not generated on 'rayextract terrain'

### DISCUSSION THREAD

**Discussion by poornima-s10** [csiro-robotics/raycloudtools#5] [Q&A] [answered]
Hi,
Thanks for making these tools available. I started using raycloudtools recently. I updated to the latest build, and now ‘rayextract terrain’ on a decimated ply does not generate a mesh. 
I have uploaded the ply file and mesh [results from a previous build](https://drive.google.com/drive/folders/1lYeW8kffLGykk1PPqI-aIpubVmtR_WET?usp=sharing) (March 2023). Now when I run the same command (rayextract terrain cloud*_decimated.ply) there is no error but a mesh is not generated. And ‘rayextract trees’ still needs a raycloud and the mesh as inputs, which is the next step I’d be running.
I wanted to check if anything relevant has changed with rayextract before I look into this further. Appreciate any pointers. Thanks again.
Poornima

**Comment by thomas-lowe** [comment, marked as answer]
Hi Poornima, sorry for the slow reply, I have just figured out how to get github to send me an e-mail notifications (I need to 'watch' my own repo). 

There were two changes since March, the first added colour to the terrain, it seems unlikely that this would cause your problem.
The second added an openmp parallel for pragma. As far as I understand, if you don't have openmp then the pragma does nothing. 

I suggest removing this #pragma (ln 180 rayterrain.cpp) and see if it fixed the problem. There may be some sort of race condition that only occurs on that particular dataset.

**Comment by sjshumack** [comment]
Hi, 

Just in case anyone else comes looking for a solution: 

I encountered a similar issue of rayextract terrain not finishing and not saving an output. In my case it turned out to be because I had **not** installed qhull and built raycloud tools using cmake .. -DWITH_QHULL=ON (see instructions in the optional build dependencies section of README.md under the heading "For raywrap:").

Thanks for releasing this excellent repo Thomas.

**Reply by ghost** [reply]
Thanks Sam, I have added a message to let the user know they need to install QHull, if they run rayextract terrain without it.
